wallet-new/src/Cardano/Wallet/Kernel/DB/Sqlite.hs

```diff
@@ -317,10 +350,12 @@ unsafeMigrateMetaDB conn =
 -- | Simply a conveniency wrapper to avoid 'Kernel.TxMeta' to explicitly
 -- import Sqlite modules.
 newConnection :: FilePath -> IO Sqlite.Connection
-newConnection = Sqlite.open
+newConnection path = Sqlite.open path
+-- TODO: add a debug option for this: Sqlite.setTrace conn (Just putStrLn)
 
 -- | Closes an open 'Connection' to the @Sqlite@ database stored in the
 -- input 'MetaDBHandle'.
+-- Even if open failed with error, this function should be called http://www.sqlite.org/c3ref/open.html
```

Comment:
Would it be better to not export `closeMetaDB` at all but rather only expose a bracket-style initializer, so that we enforce this invariant?

Reply:
@edsko Need to think a bit about it and how it interacts with migration. Would that be ok with a TODO for now?

Reply:
Yup.